hawkbit-repository/hawkbit-repository-api/src/test/java/org/eclipse/hawkbit/repository/RepositoryManagementMethodPreAuthorizeAnnotatedTest.java

```diff
@@ -0,0 +1,134 @@
+        final List<Class<?>> findInterfacesInPackage = findInterfacesInPackage(getClass().getPackage(),
+                Pattern.compile(".*Management"));
+
+        assertThat(findInterfacesInPackage).isNotEmpty();
+        for (final Class<?> interfaceToCheck : findInterfacesInPackage) {
+            assertDeclaredMethodsContainsPreAuthorizeAnnotaions(interfaceToCheck);
+        }
+
+        // all exclusion should be used, otherwise the method exlusion should be
+        // cleaned up again
+        assertThat(METHOD_SECURITY_EXCLUSION).isEmpty();
+    }
+
+    /**
+     * asserts that the given methods are annotated with the
+     * {@link PreAuthorize} annotation for security. Inherited methods are not
+     * checked. The following methods are excluded due inherited from
+     * {@link Object}, like equals() or toString().
+     * 
+     * @param clazz
+     *            the class to retrieve the public declared methods
+     */
+    private static void assertDeclaredMethodsContainsPreAuthorizeAnnotaions(final Class<?> clazz) {
+        final Method[] declaredMethods = clazz.getDeclaredMethods();
+        for (final Method method : declaredMethods) {
+            final boolean methodExcluded = METHOD_SECURITY_EXCLUSION.contains(method);
+            if (methodExcluded || method.isSynthetic() || Modifier.isPublic(method.getModifiers())) {
+                // skip method because it should be excluded
+                METHOD_SECURITY_EXCLUSION.remove(method);
+                continue;
+            }
+            final PreAuthorize annotation = method.getAnnotation(PreAuthorize.class);
+            assertThat(annotation).as("The public method " + method.getName() + " in class " + clazz.getName()
+                    + " is not annoated with @PreAuthorize, security leak?").isNotNull();
+        }
+    }
+
+    /**
+     * Finds all interfaces in a given packages which matches the given filter.
+     * 
+     * @param p
+     *            the package to search for interfaces in
+     * @param includeFilter
+     *            the pattern which interfaces class names should be included
+     * @return a list of loaded interfaces in a specific package and matches the
+     *         given filter
+     * @throws URISyntaxException
+     * @throws IOException
+     * @throws ClassNotFoundException
+     */
+    private List<Class<?>> findInterfacesInPackage(final Package p, final Pattern includeFilter)
+            throws URISyntaxException, IOException, ClassNotFoundException {
+        final List<Class<?>> interfacesToReturn = new ArrayList<>();
+        final ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+        final Enumeration<URL> resources = classLoader.getResources(p.getName().replace(".", "/"));
+        while (resources.hasMoreElements()) {
+            final String uriPath = new URI(resources.nextElement().toString()).getPath();
+            if (uriPath != null) {
```

Comment:
Line 105 - 123 is hard to read and understand. I would prefer more methods, then you can avoid the nesting ifs and for.